src/analysis/special_functions/arsinh.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2020 James Arthur. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: James Arthur, Chris Hughes, Shing Tak Lam.
+-/
+import analysis.special_functions.trigonometric
+
+open real
+noncomputable theory
+
+/-!
+# Inverse of the sinh function
+
+In this file we prove that sinh is bijective and hence has an
+inverse, arsinh.
+
+## Main Results
+
+- `sinh_injective`: The proof that `sinh` is injective
+- `sinh_surjective`: The proof that `sinh` is surjective
+- `sinh_bijective`: The proof `sinh` is bijective
+- `arsinh`: The inverse function of `sinh`
+
+## Tags
+
+arsinh, arcsinh, argsinh, asinh, sinh injective, sinh bijective, sinh surjective
+-/
+
+
+/-- `arsinh` is defined using a logarithm, `arsinh x = log (x + sqrt(1 + x^2))` -/
+def arsinh (x : ℝ) := log (x + sqrt(1 + x^2))
+
+/-- `sinh` is injective, `∀ a b, sinh a = sinh b → a = b` -/
+lemma sinh_injective : function.injective sinh := sinh_strict_mono.injective
+
+private lemma aux_lemma (x : ℝ) : 1 / (x + sqrt (1 + x^2)) = -x + sqrt (1 + x^2) :=
+begin
+  refine (eq_one_div_of_mul_eq_one _).symm,
+  have : 0 ≤ 1 + x ^ 2 := add_nonneg zero_le_one (pow_two_nonneg x),
+  rw [add_comm, ← sub_eq_neg_add, ← mul_self_sub_mul_self,
+      mul_self_sqrt this, pow_two, add_sub_cancel]
+end
+
+private lemma b_lt_sqrt_b_sq_add_one (b : ℝ) : b < sqrt (b^2 + 1) :=
+begin
+  by_cases hb : 0 ≤ b,
+  conv { to_lhs, rw ← sqrt_sqr hb },
+  rw sqrt_lt,
+  linarith,
+  apply pow_two_nonneg,
+  have F : 0 ≤ b^2 := by apply pow_two_nonneg,
+  { linarith },
+  rw not_le at hb,
+  apply lt_of_lt_of_le hb,
+  apply sqrt_nonneg,
+end
+
+/-- `arsinh` is the right inverse of `sinh`-/
+lemma arsinh_sinh (x : ℝ) : sinh (arsinh x) = x :=
+begin
+  rw sinh_eq,
+  unfold arsinh,
+  rw [←log_inv, exp_log, exp_log],
+  { rw [← one_mul ((x + sqrt (1 + x ^ 2))⁻¹), ←division_def, aux_lemma],
+    ring },
+  { rw [← one_mul ((x + sqrt (1 + x ^ 2))⁻¹), ←division_def, aux_lemma],
+    ring,
+    rw [neg_add_eq_sub, sub_pos],
+    have H : x^2 < sqrt (x ^ 2 + 1)^2,
+    { rw sqr_sqrt,
+      linarith,
+      have F : 0 ≤ x^2 := by apply pow_two_nonneg,
+      linarith },
+    exact b_lt_sqrt_b_sq_add_one x,
+  },
+  { have H := b_lt_sqrt_b_sq_add_one (-x),
+    rw [neg_square, add_comm (x^2)] at H,
+    linarith },
+end
```

Comment:
```suggestion
private lemma add_sqrt_one_add_pow_two_pos (b : ℝ) : 0 < b + sqrt (1 + b ^ 2) :=
by { rw [← neg_neg b, ← sub_eq_neg_add, sub_pos, pow_two, neg_mul_neg, ← pow_two],
  exact b_lt_sqrt_b_one_add_sq (-b) }

/-- `arsinh` is the right inverse of `sinh`-/
lemma arsinh_sinh (x : ℝ) : sinh (arsinh x) = x :=
by rw [sinh_eq, arsinh, ← log_inv, exp_log (add_sqrt_one_add_pow_two_pos x),
      exp_log (inv_pos.2 (add_sqrt_one_add_pow_two_pos x)),
      inv_eq_one_div, aux_lemma x, sub_eq_add_neg, neg_add, neg_neg, ← sub_eq_add_neg,
      add_add_sub_cancel, add_self_div_two]
```

Reply:
It won't allow me to commit suggestion, so I've done a seperate commit